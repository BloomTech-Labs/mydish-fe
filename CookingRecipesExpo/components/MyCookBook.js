import React, {useState, useEffect} from "react";
import 
{View,TouchableOpacity, TextInput, Button, StyleSheet, Text, Image} 
from "react-native";
import RecipeList from "./RecipeList";
// import AxiosWithAuth from "./";



const MyCookBook = (props) =>{
    
    const courses= ['Breakfast','Brunch','Lunch','Dinner','Dessert','Snack'];


    // useEffect(() =>{
    //     axios
    //     .get(
    //       `https://recipeshare-development.herokuapp.com/recipes?title=${dish}`
    //     )
    //     .then(res => {
    //         setRecipes([])
    //         setRecipes(res.data);
    //     })
    //     .catch(err => console.log(err));

    // },[dish]);


    return(
        <View style={ {alignSelf:'center', flexDirection: "column"}}>
        	{/* <TextInput
					style={styles.textInput}
					placeholder="What are you looking for?"
					placeholderTextColor="#D3D3D3"
					value={dish}
                    onChangeText={dish => setDish(dish)}
				/> */}
            <Text style={{fontSize: 20,fontWeight: 'bold', alignSelf: 'center', marginBottom: 5, color:`#3BA405`}}>Your Personal CookBook!</Text>
            {courses.map(cour =>{
                return(
                <TouchableOpacity onPress={()  =>  props.navigation.navigate('Courses', {Course: cour})}>
                <Text style={{margin:"20%", alignSelf:"center", fontSize: 15}}>
                    {cour}
                </Text>
                </TouchableOpacity>
                )
            })}
            
            {/* <Text>{user.props.name}</Text> */}
            {/* {folderName.props.map(folder  => {
                return (
                    <TouchableOpacity style={styles.button}>
                    <Butto n  
                    color="white"    
                    onPress={()  =>  props.navigation.navigate('CookBookFolder',  {data: folder})}
                    title={folder.courses}
                    accessibilityLabel="Search"                   
                    />
                </TouchableOpacity>
                )
            })} */}
            {/* {savedRecipes.length>=1  && <RecipeList props={savedRecipes} /> */} 
        </View>
    )

}

export default MyCookBook;