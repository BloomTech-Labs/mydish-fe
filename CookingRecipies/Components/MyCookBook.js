import React, {useState, useEffect} from "react";
import 
{View,TouchableOpacity, TextInput, Button, StyleSheet, Text} 
from "react-native";


const MyCookBook = (props) =>{
    const[folderName, setFolderName] = useState([
        {
            course: '',
            recipes:  []
        }
    ]);
    const [savedRecipes,  setSavedRecipes] = useState([props]);

    useEffect(()=>{
        for(i in savedRecipes){
            for(x in savedRecipses){
                if(i.course == folderName.course){
                    setFolderName(...folderName, {...folderName.course, recpies: recpies.append(i)})
                }
                setFolderName(...folderName, {course: i.course, recpies: recpies.append(i)})
            }
        }

    },[]);

    return(
        <View>
            {/* <Text>{user.props.name}</Text> */}
            {folderName.props.map(folder  => {
                return (
                    <TouchableOpacity style={styles.button}>
                    <Button  
                    color="white"    
                    onPress={()  =>  props.navigation.navigate('CookBookFolder',  {data: folder})}
                    title={folder.courses}
                    accessibilityLabel="Search"                   
                    />
                </TouchableOpacity>
                )
            })}
        </View>
    )

}

export default MyCookBook;