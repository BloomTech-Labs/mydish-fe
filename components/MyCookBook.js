import React, { useEffect } from 'react';
import { View, ScrollView, Text, SafeAreaView } from 'react-native';
import { withNavigation } from 'react-navigation';
import { cookbookHeaderOptions } from './header/navigationHeader';
import { getAllCookbookRecipes } from '../store/cookbook/cookbookAction';
import { useDispatch, useSelector } from 'react-redux';
import styles from '../styles/recipe-styles';
import Recipe from './Recipe';
import FancySpinner from './FancySpinner';
import AddRecipeButton from './AddRecipeButton';

const MyCookBook = ({ navigation }) => {
  const dispatch = useDispatch();
  const loading = useSelector((state) => state.cookbook.isLoading);
  const allCookbookRecipes = useSelector(
    (state) => state.cookbook.entireCookbook
  );

  useEffect(() => {
    dispatch(getAllCookbookRecipes());
  }, [dispatch, getAllCookbookRecipes]);

  const getAllCategories = (allRecipes) => {
    let categoryList = [];
    if (allRecipes) {
      allRecipes.forEach((recipe, _) => {
        if (recipe.tags && recipe.tags[0]) {
          if (!categoryList.includes(recipe.tags[0].name)) {
            categoryList = [...categoryList, recipe.tags[0].name];
          }
        }
      });
    }
    return categoryList;
  };
  const categories = getAllCategories(allCookbookRecipes);

  const cookbookHeadText = () => (
    <View style={styles.cookbookHeadContainer}>
      <Text style={styles.cookbookHeadText}>My Cookbook</Text>
      <AddRecipeButton navigation={navigation} />
    </View>
  );

  const noCookbookRecipes = () => (
    <View style={styles.noRecipeCookbookContainer}>
      {cookbookHeadText()}
      <Text style={styles.noRecipes}>
        You don't have any recipes saved yet.
      </Text>
      <AddRecipeButton navigation={navigation} />
    </View>
  );

  if (loading) return <FancySpinner />;
  if (!loading && allCookbookRecipes.length === 0) return noCookbookRecipes();

  return (
    <SafeAreaView
      style={{
        maxWidth: '90%',
        marginHorizontal: 16,
      }}
    >
      {cookbookHeadText()}
      <ScrollView>
        {categories.map((tag) => {
          return (
            <View key={tag}>
              <Text style={styles.heading}>{tag}</Text>
              {allCookbookRecipes
                .filter((recipeToFilter) => {
                  return recipeToFilter.tags[0].name === tag;
                })
                .map((filteredRecipe) => {
                  const id = filteredRecipe.id;
                  const tag = filteredRecipe.tags[0].name;
                  return (
                    <Recipe
                      key={`${id}.${tag}`}
                      recipe={filteredRecipe}
                      parent={'Cookbook'}
                    />
                  );
                })}
            </View>
          );
        })}
        <View style={{ height: 200 }} />
      </ScrollView>
    </SafeAreaView>
  );
};

MyCookBook.navigationOptions = cookbookHeaderOptions;

export default withNavigation(MyCookBook);
