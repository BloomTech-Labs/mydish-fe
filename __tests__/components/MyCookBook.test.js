import 'react-native';
import React from 'react';
import MyCookBook from '../../components/MyCookBook';
import renderer from 'react-test-renderer';

it('matches snapshot', () => {
  const tree = renderer.create(<MyCookBook />).toJSON();
  expect(tree).toMatchSnapshot();
});
