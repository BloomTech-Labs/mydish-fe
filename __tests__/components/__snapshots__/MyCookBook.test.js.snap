// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<View
  style={
    Object {
      "marginLeft": "5%",
      "width": "90%",
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "paddingBottom": "10%",
      }
    }
  >
    <View>
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "height": 200,
                "marginBottom": "15%",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": "2%",
                }
              }
            >
              Breakfast
            </Text>
            <Image
              source={
                Object {
                  "uri": "https://d9hyo6bif16lx.cloudfront.net/live/img/production/detail/menu/breakfast_breakfast-classics_big-two-do-breakfast.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 4,
                  "height": 200,
                  "paddingRight": 20,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "height": 200,
                "marginBottom": "15%",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": "2%",
                }
              }
            >
              Brunch
            </Text>
            <Image
              source={
                Object {
                  "uri": "https://media.timeout.com/images/105500044/1024/576/image.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 4,
                  "height": 200,
                  "paddingRight": 20,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "height": 200,
                "marginBottom": "15%",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": "2%",
                }
              }
            >
              Lunch
            </Text>
            <Image
              source={
                Object {
                  "uri": "https://hips.hearstapps.com/del.h-cdn.co/assets/17/41/1600x1600/square-1507827786-buddha-bowls-delish-1.jpg?resize=640:*",
                }
              }
              style={
                Object {
                  "borderRadius": 4,
                  "height": 200,
                  "paddingRight": 20,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "height": 200,
                "marginBottom": "15%",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": "2%",
                }
              }
            >
              Dinner
            </Text>
            <Image
              source={
                Object {
                  "uri": "https://img1.cookinglight.timeinc.net/sites/default/files/styles/4_3_horizontal_-_900x675/public/image/2016/09/main/_1501p108-weeknight-lemon-chicken-skillet-dinner.jpg?itok=sGWzw71z",
                }
              }
              style={
                Object {
                  "borderRadius": 4,
                  "height": 200,
                  "paddingRight": 20,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "height": 200,
                "marginBottom": "15%",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": "2%",
                }
              }
            >
              Dessert
            </Text>
            <Image
              source={
                Object {
                  "uri": "https://cdn3.tmbi.com/toh/GoogleImages/exps19201_RDS011700016SC03_13_2b_WEB.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 4,
                  "height": 200,
                  "paddingRight": 20,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "height": 200,
                "marginBottom": "15%",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": "2%",
                }
              }
            >
              Snack
            </Text>
            <Image
              source={
                Object {
                  "uri": "https://data.thefeedfeed.com/recommended/post_4483824.jpeg",
                }
              }
              style={
                Object {
                  "borderRadius": 4,
                  "height": 200,
                  "paddingRight": 20,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
